elliottlib/cli/verify_attached_operators_cli.py

```diff
@@ -0,0 +1,149 @@
+    available_shasums = _extract_available_image_shasums(image_builds)
+    if _any_references_are_missing(referenced_specs, available_shasums):
+        raise ElliottFatalError("Some references were missing. Ensure all bundle references are shipped or shipping.")
+    green_print("All operator manifest references were found.")
+
+
+def _get_attached_image_builds(brew_session, advisories):
+    # get all attached image builds
+    build_nvrs = []
+    try:
+        for advisory in advisories:
+            green_print(f"Retrieving builds from advisory {advisory}")
+            advisory = Erratum(errata_id=advisory)
+            for build_list in advisory.errata_builds.values():  # one per product version
+                build_nvrs.extend(build_list)
+    except GSSError:
+        exit_unauthenticated()
+
+    green_print(f"Found {len(build_nvrs)} builds")
+    return [build for build in brew.get_build_objects(build_nvrs, brew_session) if _is_image(build)]
+
+
+def _is_image(build):
+    return build.get('extra', {}).get('osbs_build', {}).get('kind') == "container_build"
+
+
+def _is_bundle(image_build):
+    return 'operator_bundle' in image_build.get('extra', {}).get('osbs_build', {}).get('subtypes', [])
+
+
+def _is_appregistry(image_build):
+    return 'operator_appregistry' in image_build.get('extra', {}).get('osbs_build', {}).get('subtypes', [])
+
+
+def _extract_operator_manifest_image_references(image_builds):
+    # extract referenced images from bundles to be shipped
+    # returns a map[pullspec: bundle_nvr]
+    image_specs = {}
+    for image in image_builds:
+        if _is_bundle(image):
+            for pullspec in image['extra']['image']['operator_manifests']['related_images']['pullspecs']:
+                image_specs[pullspec['new']] = image['nvr']
+        elif _is_appregistry(image):
+            for pullspec in _download_appregistry_image_references(image):
+                image_specs[pullspec] = image['nvr']
+    return image_specs
+
+
+def _download_appregistry_image_references(appregistry_build):
+    # for appregistry, image references are buried in the CSV in an archive
+    url = constants.BREW_DOWNLOAD_TEMPLATE.format(
+        name=appregistry_build['package_name'],
+        version=appregistry_build['version'],
+        release=appregistry_build['release'],
+        file_path="operator-manifests/operator_manifests.zip",
+    )
+    try:
+        res = requests.get(url, timeout=10.0)
+    except Exception as ex:
+        raise ElliottFatalError(f"appregistry data download {url} failed: {ex}")
+    if res.status_code != 200:
+        raise ElliottFatalError(f"appregistry data download {url} failed (status_code={res.status_code}): {res.text}")
+
+    minor_version = re.match(r'v(\d+\.\d+)', appregistry_build['version']).groups()[0]
+    csv = {}
+    with ZipFile(BytesIO(res.content)) as z:
+        for filename in z.namelist():
+            if re.match(f"{minor_version}/.*clusterserviceversion.yaml", filename):
```

Comment:
Might want to anchor this to start and end of string (or use `filename.startswith(minor_version) && filename.endswith(clusterserviceversion.yaml)`)

Reply:
`re.match` [does anchor to the beginning of the string](https://docs.python.org/3/library/re.html#search-vs-match) (which i've been burned by enough times to remember) but it would not hurt to be explicit about it.